### ISSUE TITLE
dataview.buffer.subarray isn't a function

### ISSUE THREAD

**Issue by ibsusu** [GoogleChromeLabs/buffer-backed-object#20]
Utilizing the ArrayBuffer portion of the BufferBackedObject API chokes on this:
https://github.com/GoogleChromeLabs/buffer-backed-object/blob/e4911c3252e8f39ac50af2dcf389022ba315db8f/buffer-backed-object.js#L166

And I think it's because dataView.buffer is an ArrayBuffer.  From what I've found ArrayBuffer does not have a subarray property.  TypedArrays do though.

**Comment by ibsusu** [comment]
Here are some upgrades(?) that you can implement if you like.
1. in the get property of the Proxy object that is created I added these two helper functions.  Not sure if I'm wasting stuff or not.
```js
if(propName === "size"){
        function sizer(keyName){
          return descriptors[keyName].size;
        }
        return sizer;
      }
      if(propName === 'descriptors'){
        let descs = {};
        for(let key in descriptors){
          descs[key] = descriptors[key].size;
        }
        return descs;
      }
```

2. Change ArrayBuffer to return a Uint8Array since we can grab the buffer.
```js
BufferBackedObject.ArrayBuffer = (size) => ({
  size,
  get: (dataView, byteOffset) =>
    new Uint8Array(dataView.buffer, byteOffset, size),
  set: (dataView, byteOffset, value) =>
    new Uint8Array(dataView.buffer, byteOffset, size).set(
      new Uint8Array(value)
    ),
});
```

3.  Add some sort of bytewise access for the object properties when they're being defined.
```js
Object.defineProperty(target[idx], name+'Data', {
            enumerable: false,
            get() {
              return new Uint8Array(dataView.buffer, itemOffset+descriptor.offset, descriptor.size);
            }
          });
```
4.  I haven't figured out how to do the sorting from inside of the Proxy Object.  I'm just going to do it outside of it and split it among workers.


Thanks for this cool library.  Sorry about the lack of prs, I don't like signing stuff.

**Comment by surma** [comment]
> Sorry about the lack of prs, I don't like signing stuff.

I understand. Don’t worry about it.

Yeah that was an oversight by me (and something I didn’t have a test for). I think I’ll just remove `ArrayBuffer` in favor of `Uint8` et al.

**Comment by surma** [comment]
Closed via d31833a
